File: game-server/src/main/java/work/fking/pangya/game/player/Caddie.java
package work.fking.pangya.game.player;

import io.netty.buffer.ByteBuf;
import work.fking.pangya.game.model.IffObject;

public record Caddie(
        int uid,
        int iffId,
        int levelsGained,
        int experience
) implements IffObject {

    public static Caddie mock() {
        return new Caddie(83651, 469762083, 2, 123);
    }

    @Override
    public void encode(ByteBuf buffer) {
        buffer.writeIntLE(uid);
        buffer.writeIntLE(iffId);
        buffer.writeIntLE(0);
        buffer.writeByte(levelsGained);
        buffer.writeIntLE(experience);
        buffer.writeLongLE(0);
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/player/Caddie.kt
package work.fking.pangya.game.player

import io.netty.buffer.ByteBuf
import work.fking.pangya.game.model.IffObject

data class Caddie(
    override val uid: Int = -1,
    override val iffId: Int,
    val level: Int = 0,
    val experience: Int = 0
) : IffObject {

    override fun encode(buffer: ByteBuf) {
        buffer.writeIntLE(uid)
        buffer.writeIntLE(iffId)
        buffer.writeIntLE(0)
        buffer.writeByte(level)
        buffer.writeIntLE(experience)
        buffer.writeLongLE(0)
    }
}

fun nullCaddie() = Caddie(iffId = 0)
